### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by MarenLeisner**
Release Notes

    New Features
        Introduced new scripts for managing MLflow experiments, logging, and registering machine learning models in a Databricks environment.
        Added functionality for saving processed datasets and enhanced data visualization capabilities.
        Implemented a configuration class for better settings management.
        Added new methods for data processing, including splitting datasets and saving to a catalog.

    Chores
        Updated project metadata and dependencies in pyproject.toml for improved compatibility and performance.
        Introduced a CI workflow step for installing the pre-commit package.

## Release Notes

- **New Features**
  - Introduced a comprehensive logging mechanism and enhanced data processing workflow.
  - Added functionality for managing MLflow experiments and integrating with Databricks.
  - Implemented a machine learning pipeline for fare predictions using LightGBM.
  - New visualization functions for model evaluation and feature importance.
  - Added configuration management for machine learning projects via YAML files.

- **Bug Fixes**
  - Improved error handling for configuration loading and data processing tasks.

- **Documentation**
  - Updated `README.md` for formatting consistency.

- **Chores**
  - Updated project dependencies and configuration files for better management.
  - Added new entries to `.gitignore` to exclude Databricks-related files.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    raise
+except yaml.YAMLError as e:
+    logger.error("Error parsing configuration file: %s", e)
+    raise
+
+spark = DatabricksSession.builder.profile(os.environ["DATABRICKS_PROFILE"]).getOrCreate()
+
+try:
+    path = (
+        config.read_from["catalog_name"] + "." + config.read_from["schema_name"] + "." + config.read_from["table_name"]
```
Comment (mvechtomova) [inline main.py]: better to use f-string, like f"{config.read_from["catalog_name"]}.{config.read_from["schema_name"]}.{config.read_from["table_name"]}"

File: notebooks/week2/01_prepare_dataset_n.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# COMMAND ----------
+
+# path stuff is annoying, in vs code it seems like everything gets executed from repo main level
+try:
+    config = ProjectConfig.from_yaml(config_path="project_config.yml")
+except Exception:
+    config = ProjectConfig.from_yaml(config_path="../../project_config.yml")
+
+spark = DatabricksSession.builder.profile(os.environ["DATABRICKS_PROFILE"]).getOrCreate()
```
Comment (mvechtomova) [inline notebooks/week2/01_prepare_dataset_n.py:23]: better to use SparkSession, and no profile definition - this code should run anywhere

File: notebooks/week2/01_prepare_dataset_n.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+try:
+    path = (
+        config.read_from["catalog_name"] + "." + config.read_from["schema_name"] + "." + config.read_from["table_name"]
```
Comment (mvechtomova) [inline notebooks/week2/01_prepare_dataset_n.py]: f-string is nicer

File: notebooks/week2/05_log_and_register_fe_model.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
+# Databricks notebook source
+# MAGIC %pip install ../power_consumption-0.0.1-py3-none-any.whl
```
Comment (mvechtomova) [inline notebooks/week2/05_log_and_register_fe_model.py]: your use case is taxi, right?

File: pyproject.toml [hunk 5]
```diff
@@ -1,27 +1,25 @@
 [project]
-name = "mlops-with-databricks"
+name = "house-price"
```
Comment (mvechtomova) [inline pyproject.toml]: it is taxi, right?

**Review by mvechtomova — COMMENTED** [review]
Some cleaning is required. Please remove mlruns and put it in gitignore as well.

File: main.py (hunk 1 above)
Comment (MarenLeisner) [reply main.py]: I fixed that

File: notebooks/week2/05_log_and_register_fe_model.py (hunk 4 above)
Comment (MarenLeisner) [reply notebooks/week2/05_log_and_register_fe_model.py]: i commented it out. but yes. i fixed it anyway

File: notebooks/week2/01_prepare_dataset_n.py (hunk 3 above)
Comment (MarenLeisner) [reply notebooks/week2/01_prepare_dataset_n.py]: fixed this aswell

File: notebooks/week2/05_log_and_register_fe_model.py (hunk 4 above)
Comment (MarenLeisner) [reply notebooks/week2/05_log_and_register_fe_model.py]: it was a comment but i changed this

File: pyproject.toml (hunk 5 above)
Comment (MarenLeisner) [reply pyproject.toml]: Yes it was taxi. i changed this aswell

File: .gitignore [hunk 6]
```diff
@@ -95,3 +95,5 @@ dmypy.json
 # VS code configuration
```
Comment (basakeskili) [inline .gitignore:95]: you can add mlruns here to ignore mlflow run artifacts

**Review by basakeskili — APPROVED** [review]

**Pull request merged**